gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -554,23 +554,16 @@ public String toString() {
   }
 
   private IOException asIOException(StorageException oops) {
+    // RPC API can only throw StorageException, but CloudStorageFileSystemProvider
+    // can only throw IOException. Square peg, round hole.
+    // TODO: research if other codes should be translated similarly.
```

Comment:
Optional: can create a github issue and add the issue number in this TODO for tracking/discussion purposes.